torch_xla/csrc/helpers.cpp

```diff
@@ -577,4 +577,48 @@ xla::XlaOp XlaHelpers::PromotedLogicalUnaryOp(
   return unary_op(op);
 }
 
+xla::StatusOr<xla::XlaComputation> XlaHelpers::WrapXlaComputation(
+    const xla::XlaComputation& computation,
+    const std::vector<xla::Shape>& parameter_shapes,
+    std::vector<std::pair<int64_t, int64_t>> input_output_alias_pair) {
+  xla::XlaBuilder builder(computation.proto().name());
+
+  // Construct a single tuple parameter.
+  const xla::XlaOp input_tuple = [&builder, &parameter_shapes]() {
+    xla::Shape input_tuple;
+    input_tuple.set_element_type(xla::PrimitiveType::TUPLE);
+    input_tuple.mutable_tuple_shapes()->reserve(parameter_shapes.size());
+    for (int i = 0; i < parameter_shapes.size(); ++i) {
+      *input_tuple.add_tuple_shapes() = parameter_shapes[i];
+    }
+    return xla::Parameter(&builder, 0, input_tuple, "in");
+  }();
+
+  // Handle the results of the original computation.
+  const std::vector<xla::XlaOp> inner_params = [&input_tuple,
```

Comment:
Just for my education, what's the advantage of defining these parameters in such way, i.e, using lambdas?

Reply:
oh.. This part of code was modified from a legacy internal code. From what I can tell this just keeps all of the intermediate variable within the lambda scope. I will remove it as we don't use this kind of style in other part of pt/xla.

Reply:
First time seeing such technique actually. Maybe under some circumstances, it's good to free memory gradually in the scope of function instead of all at once during return... If just for readability, block should do the trick I guess?

Reply:
yea, I think so. The only thing I can think of(why this is preferred than a block) is that one must specified what variable from outer block can be accessed. I don't think it fits code style for this repo too good so I will remove it.

Reply:
That's fair. (TBH, I don't use blocks often myself...)